### PR TITLE
feat: forward amount and assetId on contract call

### PR DESCRIPTION
**Pull request by luizstacio**
[chore: update lib-core and lib-std](https://github.com/FuelLabs/fuels-ts/commit/a6dfafe450af7dc16eb33e1259f6a46e4c6ad28a)

 - Update std libs with latest changes on `msg_asset_id()` and `msg_amount()`
 
 [feat: forward asset and amount on contract call](https://github.com/FuelLabs/fuels-ts/commit/b6b64d640318e5389bd5b9faba5b4e0c08adc724)
 
 - Enable forward custom asset and amount on contract call

On typescript;
```
const result = await contract.functions.deposit({
  amount: 1_000_000,
  assetId: '0x0000000000000000000000000000000000000000000000000000000000000000',
});
```

On Sway these values can be retrieved like this;
```
  fn return_context_amount() -> u64 {
    msg_amount()
  }
  fn return_context_asset() -> b256 {
    (msg_asset_id()).into()
  }
```

closes: #198

### CODE REVIEW

File: packages/contract/src/scripts.ts [hunk 1]
```diff
@@ -9,35 +12,56 @@ import { Script, ReceiptType } from '@fuel-ts/providers';
...
+    // Load the amount into 0x12
+    Opcode::ADDI(0x12, REG_ZERO, data_offset + 32),
+    // Load word into 0x12
+    Opcode::LW(0x12, 0x12, 0),
+    // Load the asset id to use to 0x13.
+    Opcode::ADDI(0x13, REG_ZERO, data_offset),
+    // Call the transfer contract.
+    Opcode::CALL(0x10, 0x12, 0x13, REG_CGAS),
+    Opcode::RET(REG_ONE),
   */
```
Comment (AlicanC) [inline packages/contract/src/scripts.ts:32]: Is this correct? A comment says `Load gas forward to 0x11`, but I don't see `0x11` being used.

**Review by AlicanC — COMMENTED** [review]
LGTM w/ 1 comment

File: packages/contract/src/scripts.ts (hunk 1 above)
Comment (luizstacio) [reply packages/contract/src/scripts.ts:32]: This is a good point, during don't correctly the answer. cc @digorithm.

Comment (digorithm) [reply packages/contract/src/scripts.ts:32]: Oh, I forgot to remove that comment on the Rust SDK I guess? We're still not forwarding gas. Once we do we can use 0x11 (or any unreserved) for that.

**Review by AlicanC — APPROVED** [review]

**Pull request merged**